src/main/java/org/apache/commons/lang3/time/FastDatePrinter.java

```diff
@@ -162,8 +165,8 @@ private void init() {
         final List<Rule> rulesList = parsePattern();
         mRules = rulesList.toArray(new Rule[rulesList.size()]);
 
-        int len = 0;
-        for (int i=mRules.length; --i >= 0; ) {
+        @NonNegative int len = 0;
```

Comment:
Why are these annotations necessary? They're on local variables, and should therefore be inferred.